components/core/src/clp_s/search/antlr_common/ErrorListener.hpp

```diff
@@ -0,0 +1,30 @@
+#ifndef CLP_S_SEARCH_ANTLRCOMMON_ERRORLISTENER_HPP
+#define CLP_S_SEARCH_ANTLRCOMMON_ERRORLISTENER_HPP
+#include <antlr4-runtime.h>
+
+namespace clp_s::search::antlr_common {
+class ErrorListener : public antlr4::BaseErrorListener {
+public:
+    void syntaxError(
+            antlr4::Recognizer* recognizer,
+            antlr4::Token* offending_symbol,
+            size_t line,
+            size_t char_position_in_line,
+            std::string const& msg,
+            std::exception_ptr e
+    ) override {
+        m_error = true;
+        m_error_message = msg;
+    }
+
+    [[nodiscard]] auto error() const -> bool { return m_error; }
+
+    [[nodiscard]] auto message() const -> std::string const& { return m_error_message; }
```

Comment:
```suggestion
    [[nodiscard]] auto message() const -> std::string_view { return m_error_message; }
```
How about returning a string view instead?